File: Python/programmers/kakao/level1/비밀_지도.py
def solution(n, arr1, arr2):
    answer = []
    for i in range(n):
        tmp = bin(arr1[i]|arr2[i])[2:]
        tmp = tmp.zfill(n)
        tmp = tmp.replace("1", "#").replace("0", " ")
        answer.append(tmp)

    return answer

n = 5
arr1 = [9, 20, 28, 18, 11]
arr2 = [30, 1, 21, 17, 28]
print(solution(n, arr1, arr2))
# ["#####","# # #", "### #", "# ##", "#####"]

n = 6
arr1 = [46, 33, 33 ,22, 31, 50]
arr2 = [27 ,56, 19, 14, 14, 10]
print(solution(n, arr1, arr2))
# ["######", "### #", "## ##", " #### ", " #####", "### # "]

File: Java/programmers/kakao/level1/비밀지도.java
package CodeTest.Java.programmers.kakao.level1;

public class 비밀지도 {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];

        for (int i=0; i<n; i++) {
            String[] tmp = Integer.toBinaryString(arr1[i]|arr2[i]).split("");
            int t = tmp.length;
            StringBuilder stb = new StringBuilder();
            
            for (int j=0; j<n-t; j++) {
                stb.append(" ");
            }

            for (int j=0; j<t; j++) {
                if (tmp[j].equals("0")) {
                    stb.append(" ");
                } else {
                    stb.append("#");
                }
            }
            
            answer[i] = stb.toString();
        }

        return answer;
    }
}
